crates/uv-git/src/git.rs

```diff
@@ -969,13 +969,34 @@ pub(crate) fn fetch(
 
     debug!("Performing a Git fetch for: {remote_url}");
     match strategy {
-        FetchStrategy::Cli => fetch_with_cli(repo, remote_url, &refspecs, tags),
+        FetchStrategy::Cli => {
+            // Try each refspec
+            let results = refspecs
+                .into_iter()
+                .map(|refspec| fetch_with_cli(repo, remote_url, &[refspec], tags))
+                .collect::<Vec<_>>();
+
+            // Only fail if _all_ refspecs cannot be found
+            // This matches the Libgit2 behavior
+            if results.iter().all(Result::is_err) {
+                let mut combined_err = anyhow!("failed to fetch all refspecs");
+                for err in results {
+                    if let Err(err) = err {
+                        combined_err = combined_err.context(err.to_string());
+                    }
+                }
+                Err(combined_err)
+            } else {
+                Ok(())
+            }
```

Comment:
Is there a better way to combine these?

Reply:
@BurntSushi would have good intuition here (but doesn't need to block the PR).

Reply:
per https://github.com/astral-sh/uv/pull/1786/commits/acc4ff3b7f8abf5ca26a1944deb831fc3fa07612 I don't know if these are surfaced _at all_?

```bash
$ cargo run -- pip install "uv-public-pypackage @ git+https://github.com/astral-test/uv-public-pypackage@2.0.0" -v --reinstall --no-cache
...
    0.373747s DEBUG uv_git::source Updating git source `Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("github.com")), port: None, path: "/astral-test/uv-public-pypackage", query: None, fragment: None }`
    0.377127s DEBUG uv_git::git Attempting GitHub fast path for: https://api.github.com/repos/astral-test/uv-public-pypackage/commits/2.0.0
    0.624614s DEBUG uv_git::git failed to check github HTTP status client error (422 Unprocessable Entity) for url (https://api.github.com/repos/astral-test/uv-public-pypackage/commits/2.0.0)
    0.625373s DEBUG uv_git::git skipping gc as there's only 0 pack files
    0.626108s DEBUG uv_git::git Performing a Git fetch for: https://github.com/astral-test/uv-public-pypackage
    0.627057s DEBUG uv_git::git initiating fetch of ["+refs/heads/2.0.0:refs/remotes/origin/2.0.0", "+refs/tags/2.0.0:refs/remotes/origin/tags/2.0.0"] from https://github.com/astral-test/uv-public-pypackage
error: Failed to download and build: uv-public-pypackage @ git+https://github.com/astral-test/uv-public-pypackage@2.0.0
  Caused by: Git operation failed
  Caused by: failed to find branch or tag `2.0.0`
```

Reply:
Yeah this _should_ create a causal chain with each link being an error in `results`. For example:

```rust
fn main() -> anyhow::Result<()> {
    let mut err = anyhow::anyhow!("error #1");
    for i in 2..=10 {
        err = err.context(format!("error #{i}"));
    }
    Err(err)
}
```

Has this output:

```
[andrew@duff anyhow-err-combine]$ cargo run -q
Error: error #10

Caused by:
    0: error #9
    1: error #8
    2: error #7
    3: error #6
    4: error #5
    5: error #4
    6: error #3
    7: error #2
    8: error #1
```

I'm not sure if this is the right presentation though. Because these are more like sibling errors than errors in a causal chain. Perhaps if we "picked" on error and added a `tracing::warn!` for the others? Or alternatively, smushed all of the errors into one, without creating a new link in the causal chain for each one.

Reply:
I don't think these are actually displayed right now so I'd like to do the simplest thing. I'll log the errors and raise a new one.